src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,236 @@
+  lmul_comm_rmul_rmul := λ a b, by rw [mul_assoc],
+  lmul_lmul_comm_rmul := λ a b, by rw [←mul_assoc],
+  rmul_comm_rmul_rmul := λ a b, by rw [← mul_assoc, ← mul_assoc], }
+
+@[priority 100] -- see Note [lower instance priority]
+instance comm_semigroup.is_comm_jordan [comm_semigroup A] : is_comm_jordan A :=
+{ mul_comm := mul_comm,
+  lmul_comm_rmul_rmul := λ a b, mul_assoc _ _ _, }
+
+local notation `L` := add_monoid.End.mul_left
+local notation `R` := add_monoid.End.mul_right
+
+/-!
+The Jordan axioms can be expressed in terms of commuting multiplication operators.
+-/
+section commute
+variables {A} [non_unital_non_assoc_ring A] [is_jordan A]
+
+@[simp] lemma commute_lmul_rmul (a : A) : commute (L a) (R a) :=
+add_monoid_hom.ext $ λ b, (is_jordan.lmul_comm_rmul _ _).symm
+
+@[simp] lemma commute_lmul_lmul_sq (a : A) : commute (L a) (L (a * a)) :=
+add_monoid_hom.ext $ λ b, (is_jordan.lmul_lmul_comm_lmul _ _).symm
+
+@[simp] lemma commute_lmul_rmul_sq (a : A) : commute (L a) (R (a * a)) :=
+add_monoid_hom.ext $ λ b, (is_jordan.lmul_comm_rmul_rmul _ _).symm
+
+@[simp] lemma commute_lmul_sq_rmul (a : A) : commute (L (a * a)) (R a) :=
+add_monoid_hom.ext $ λ b, (is_jordan.lmul_lmul_comm_rmul _ _)
+
+@[simp] lemma commute_rmul_rmul_sq (a : A) : commute (R a) (R (a * a)) :=
+add_monoid_hom.ext $ λ b, (is_jordan.rmul_comm_rmul_rmul _ _).symm
+
+end commute
+
+variables {A} [non_unital_non_assoc_ring A] [is_comm_jordan A]
+
+/-!
+The endomorphisms on an additive monoid `add_monoid.End` form a `ring`, and this may be equipped
+with a Lie Bracket via `ring.has_bracket`.
+-/
+
+lemma two_nsmul_lie_lmul_lmul_add_eq_lie_lmul_lmul_add (a b : A) :
+  2•(⁅L a, L (a * b)⁆ + ⁅L b, L (b * a)⁆) = ⁅L (a * a), L b⁆ + ⁅L (b * b), L a⁆ :=
+begin
+  rw ← sub_eq_zero,
+  symmetry,
+  calc 0 = ⁅L (a + b), L ((a + b) * (a + b))⁆ : by rw (commute_lmul_lmul_sq (a + b)).lie_eq
+    ... = ⁅L a + L b, L (a * a + a * b + (b * a + b * b))⁆ :
+      by rw [add_mul, mul_add, mul_add, map_add]
+    ... = ⁅L a + L b, L (a * a) + L(a * b) + (L(a * b) + L(b * b))⁆ :
+      by rw [map_add, map_add, map_add, is_comm_jordan.mul_comm b a]
+    ... = ⁅L a + L b, L (a * a) + 2•L(a * b) + L(b * b)⁆ :
+      by rw [ two_smul, ← add_assoc, ← add_assoc]
+    ... = ⁅L a, L (a * a)⁆ + ⁅L a, 2•L(a * b)⁆ + ⁅L a, L(b * b)⁆
+      + (⁅L b, L (a * a)⁆ + ⁅L b, 2•L(a * b)⁆ + ⁅L b,L(b * b)⁆) :
+        by rw [add_lie, lie_add, lie_add, lie_add, lie_add]
+    ... = 2•⁅L a, L (a * b)⁆ + ⁅L a, L(b * b)⁆ + (⁅L b, L (a * a)⁆ + 2•⁅L b,L(a * b)⁆) :
+      by rw [(commute_lmul_lmul_sq a).lie_eq, (commute_lmul_lmul_sq  b).lie_eq,
+        lie_nsmul, lie_nsmul, zero_add, add_zero]
+    ... = 2•⁅L a, L (a * b)⁆ + 2•⁅L b, L (a * b)⁆ + ⁅L a, L (b * b)⁆ + ⁅L b, L (a * a)⁆ : by abel
+    ... = 2•(⁅L a, L (a * b)⁆ + ⁅L b, L (b * a)⁆) - (⁅L (a * a), L b⁆ + ⁅L (b * b), L a⁆) : by
+      rw [← nsmul_add, sub_add_eq_sub_sub_swap, sub_eq_add_neg, lie_skew, sub_eq_add_neg,
+       lie_skew, is_comm_jordan.mul_comm b a]
```

Comment:
For what it's worth, I think the following proof is easier to grasp as well as shorter:
```suggestion
  suffices : 2 • ⁅L a, L (a * b)⁆ + 2 • ⁅L b, L (a * b)⁆ + ⁅L b, L (a * a)⁆ + ⁅L a, L (b * b)⁆ = 0,
  { rwa [← sub_eq_zero, ← sub_sub, sub_eq_add_neg, sub_eq_add_neg, lie_skew, lie_skew,
      is_comm_jordan.mul_comm b a, nsmul_add], },
  convert (commute_lmul_lmul_sq (a + b)).lie_eq,
  simp only [add_mul, mul_add, map_add, lie_add, add_lie, zero_add, add_zero,
    (commute_lmul_lmul_sq a).lie_eq, (commute_lmul_lmul_sq b).lie_eq, is_comm_jordan.mul_comm b a],
  abel,
```
I suspect `two_nsmul_lie_lmul_lmul_add_add_eq_zero` could also be golfed a lot.

Reply:
Thanks. I suspect that you're right about `two_nsmul_lie_lmul_lmul_add_add_eq_zero`. I may come back and look at that later.